src/workerd/server/server.c++

```diff
@@ -3087,18 +3087,55 @@ kj::Own<Server::Service> Server::makeWorker(kj::StringPtr name,
   auto observer = kj::atomicRefcounted<IsolateObserver>();
   auto limitEnforcer = kj::refcounted<NullIsolateLimitEnforcer>();
 
+  // Create the FsMap that will be used to map known file system
+  // roots to configurable locations.
+  // TODO(node-fs): This is set up to allow users to configure the "mount"
+  // points for known roots but we currently do not expose that in the
+  // config. So for now this just uses the defaults.
+  auto fsMap = FsMap::createDefault();
+  kj::Own<kj::Filesystem> workerFs = getWorkerFileSystem(*fsMap, getBundleDirectoryDelegate(conf));
+
+  // This callback is used to resolve file system URLs to the actual
+  // underlying file system nodes.
+  auto fsMapProvider = [workerFs = kj::mv(workerFs)](jsg::Lock&,
+                           const jsg::Url& url) -> kj::Maybe<kj::Own<const kj::FsNode>> {
+    kj::Path root{};
+    kj::String pathname = kj::str(url.getPathname().slice(1));
+    kj::Path path = root.eval(pathname);
+
+    // First, try to find it as a file.
+    KJ_IF_SOME(file, workerFs->getRoot().tryOpenFile(path)) {
```

Comment:
This means you're iterating the list of entries twice, seems like this might be a clue that the real API we need is `tryOpenNode` that returns a OneOf.